[Bug] Maven deployment failed on project pulsar-presto-connector-original

### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Version

Pulsar branch-2.8

```
$ mvn --version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /Users/xxx/.sdkman/candidates/maven/current
Java version: 11.0.16, vendor: Amazon.com Inc., runtime: /Users/xxx/.sdkman/candidates/java/11.0.16-amzn
Default locale: zh_CN_#Hans, platform encoding: UTF-8
OS name: "mac os x", version: "12.3.1", arch: "x86_64", family: "mac"
```

### Minimal reproduce step

It has succeeded before. But it failed twice in a row when I tried to deploy it again. I have already removed the staging repository. See

<img width="1076" alt="image" src="https://user-images.githubusercontent.com/18204803/183945368-4572b5e7-0db2-4fb3-a8f3-2bf03fe9c695.png">

Follow https://github.com/apache/pulsar/wiki/Release-process#5-stage-artifacts-in-maven (Apache Committer is required)

```bash
mvn clean install -DskipTests

export APACHE_USER=$USER
export APACHE_PASSWORD=$MY_PASSWORD
export GPG_TTY=$(tty)
curl -s -o /tmp/mvn-apache-settings.xml https://raw.githubusercontent.com/apache/pulsar/master/src/settings.xml
mvn deploy -DskipTests -Papache-release --settings /tmp/mvn-apache-settings.xml
```

### What did you expect to see?

`mvn deploy` should succeed.

### What did you see instead?

Recently I encounted a problem when I ran the `mvn deploy` step in
https://github.com/apache/pulsar/wiki/Release-process#5-stage-artifacts-in-maven.

> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-shade-plugin:3.2.4:shade (default) on project pulsar-presto-connector-original: Error creating shaded jar: duplicate entry: META-INF/services/org.apache.pulsar.shade.org.glassfish.jersey.internal.inject.InjectionManagerFactory -> [Help 1]

```
[INFO] pulsar-broker-auth-athenz .......................... SUCCESS [ 17.252 s]
[INFO] pulsar-client-auth-athenz .......................... SUCCESS [ 17.002 s]
[INFO] Pulsar SQL :: Parent ............................... SUCCESS [  4.238 s]
[INFO] Pulsar SQL :: Pulsar Presto Connector Packaging .... FAILURE [  6.922 s]
[INFO] Pulsar SQL :: Pulsar Presto Connector .............. SKIPPED
[INFO] Pulsar SQL :: Java Version Trim Agent .............. SKIPPED
[INFO] Pulsar SQL :: Pulsar Presto Distribution ........... SKIPPED
```

We can see the previous artifacts were deployed successfully.

### Anything else?

There were warning logs about the "overlapping resource"

```
[WARNING] aopalliance-repackaged-2.6.1.jar, hk2-api-2.6.1.jar, hk2-locator-2.6.1.jar, hk2-utils-2.6.1.jar, jakarta.inject-2.6.1.jar, javax.annotation-api-1.3.2.jar, javax.ws.rs-api-2.1.jar, jersey-client-2.34.jar, jersey-common-2.34.jar, jersey-entity-filtering-2.34.jar, jersey-hk2-2.34.jar, jersey-media-json-jackson-2.34.jar, jersey-media-multipart-2.34.jar, jetty-io-9.4.48.v20220622.jar, jetty-security-9.4.27.v20200227.jar, jetty-server-9.4.48.v20220622.jar, jetty-servlet-9.4.48.v20220622.jar, jetty-util-9.4.48.v20220622.jar, jetty-util-ajax-9.4.48.v20220622.jar, managed-ledger.jar, osgi-resource-locator-1.0.3.jar, pulsar-client-admin-original.jar, pulsar-client-original.jar, pulsar-metadata.jar, pulsar-presto-connector-original.jar define 1 overlapping resource: 
[WARNING]   - META-INF/MANIFEST.MF
[WARNING] managed-ledger.jar, pulsar-client-admin-original.jar, pulsar-client-original.jar, pulsar-metadata.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/DEPENDENCIES
[WARNING]   - META-INF/NOTICE
[WARNING] jetty-io-9.4.48.v20220622.jar, jetty-security-9.4.27.v20200227.jar, jetty-server-9.4.48.v20220622.jar, jetty-servlet-9.4.48.v20220622.jar, jetty-util-9.4.48.v20220622.jar, jetty-util-ajax-9.4.48.v20220622.jar, managed-ledger.jar, pulsar-client-admin-original.jar, pulsar-client-original.jar, pulsar-metadata.jar, pulsar-presto-connector-original.jar define 1 overlapping resource: 
[WARNING]   - META-INF/LICENSE
[WARNING] pulsar-client-admin-original.jar, pulsar-presto-connector-original.jar define 233 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-client-admin-original/pom.properties
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-client-admin-original/pom.xml
[WARNING]   - org.apache.pulsar.client.admin.internal.BaseResource
[WARNING]   - org.apache.pulsar.client.admin.internal.BaseResource$1
[WARNING]   - org.apache.pulsar.client.admin.internal.BaseResource$2
[WARNING]   - org.apache.pulsar.client.admin.internal.BaseResource$3
[WARNING]   - org.apache.pulsar.client.admin.internal.BookiesImpl
[WARNING]   - org.apache.pulsar.client.admin.internal.BookiesImpl$1
[WARNING]   - org.apache.pulsar.client.admin.internal.BookiesImpl$2
[WARNING]   - org.apache.pulsar.client.admin.internal.BookiesImpl$3
[WARNING]   - 223 more...
[WARNING] pulsar-client-original.jar, pulsar-presto-connector-original.jar define 276 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-client-original/pom.properties
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-client-original/pom.xml
[WARNING]   - findbugsExclude.xml
[WARNING]   - org.apache.pulsar.client.impl.AbstractBatchMessageContainer
[WARNING]   - org.apache.pulsar.client.impl.AcknowledgmentsGroupingTracker
[WARNING]   - org.apache.pulsar.client.impl.AuthenticationUtil
[WARNING]   - org.apache.pulsar.client.impl.AuthenticationUtil$1
[WARNING]   - org.apache.pulsar.client.impl.Backoff
[WARNING]   - org.apache.pulsar.client.impl.BackoffBuilder
[WARNING]   - org.apache.pulsar.client.impl.BatchMessageAcker
[WARNING]   - 266 more...
[WARNING] aopalliance-repackaged-2.6.1.jar, hk2-api-2.6.1.jar, hk2-locator-2.6.1.jar, hk2-utils-2.6.1.jar, jakarta.inject-2.6.1.jar, jersey-client-2.34.jar, jersey-common-2.34.jar, jersey-entity-filtering-2.34.jar, jersey-hk2-2.34.jar, jersey-media-json-jackson-2.34.jar, jersey-media-multipart-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/LICENSE.md
[WARNING]   - META-INF/NOTICE.md
[WARNING] jersey-client-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.jersey.core/jersey-client/pom.xml
[WARNING] jersey-common-2.34.jar, jersey-media-json-jackson-2.34.jar, pulsar-presto-connector-original.jar define 1 overlapping resource: 
[WARNING]   - META-INF/NOTICE.markdown
[WARNING] jersey-common-2.34.jar, pulsar-presto-connector-original.jar define 12 overlapping classes and resources: 
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.JerseyFlowSubscriber
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.JerseyFlowSubscriber$1
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$1
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$2
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$3
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$4
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$5
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisher$6
[WARNING]   - META-INF.versions.11.org.glassfish.jersey.internal.jsr166.SubmissionPublisherFactory
[WARNING]   - 2 more...
[WARNING] osgi-resource-locator-1.0.3.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2/osgi-resource-locator/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2/osgi-resource-locator/pom.xml
[WARNING] jakarta.inject-2.6.1.jar, pulsar-presto-connector-original.jar define 8 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2.external/jakarta.inject/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2.external/jakarta.inject/pom.xml
[WARNING]   - javax.inject.Inject
[WARNING]   - javax.inject.Named
[WARNING]   - javax.inject.Provider
[WARNING]   - javax.inject.Qualifier
[WARNING]   - javax.inject.Scope
[WARNING]   - javax.inject.Singleton
[WARNING] jersey-media-json-jackson-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.jersey.media/jersey-media-json-jackson/pom.xml
[WARNING] jersey-entity-filtering-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.jersey.ext/jersey-entity-filtering/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.jersey.ext/jersey-entity-filtering/pom.xml
[WARNING] jersey-media-multipart-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.jersey.media/jersey-media-multipart/pom.xml
[WARNING] jersey-hk2-2.34.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.glassfish.jersey.inject/jersey-hk2/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.jersey.inject/jersey-hk2/pom.xml
[WARNING] hk2-locator-2.6.1.jar, pulsar-presto-connector-original.jar define 101 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-locator/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-locator/pom.xml
[WARNING]   - org.jvnet.hk2.external.generator.ServiceLocatorGeneratorImpl
[WARNING]   - org.jvnet.hk2.external.runtime.Hk2LocatorUtilities
[WARNING]   - org.jvnet.hk2.external.runtime.Hk2LocatorUtilities$1
[WARNING]   - org.jvnet.hk2.external.runtime.ServiceLocatorRuntimeBean
[WARNING]   - org.jvnet.hk2.internal.AnnotatedElementAnnotationInfo
[WARNING]   - org.jvnet.hk2.internal.AutoActiveDescriptor
[WARNING]   - org.jvnet.hk2.internal.CacheKey
[WARNING]   - org.jvnet.hk2.internal.CacheKey$1
[WARNING]   - 91 more...
[WARNING] aopalliance-repackaged-2.6.1.jar, pulsar-presto-connector-original.jar define 28 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2.external/aopalliance-repackaged/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2.external/aopalliance-repackaged/pom.xml
[WARNING]   - org.aopalliance.aop.Advice
[WARNING]   - org.aopalliance.aop.AspectException
[WARNING]   - org.aopalliance.instrument.Instrumentation
[WARNING]   - org.aopalliance.instrument.InstrumentationError
[WARNING]   - org.aopalliance.instrument.Instrumentor
[WARNING]   - org.aopalliance.instrument.UndoNotSupportedException
[WARNING]   - org.aopalliance.intercept.ConstructorInterceptor
[WARNING]   - org.aopalliance.intercept.ConstructorInvocation
[WARNING]   - 18 more...
[WARNING] hk2-api-2.6.1.jar, pulsar-presto-connector-original.jar define 7 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-api/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-api/pom.xml
[WARNING]   - org.jvnet.hk2.annotations.Contract
[WARNING]   - org.jvnet.hk2.annotations.ContractsProvided
[WARNING]   - org.jvnet.hk2.annotations.InhabitantAnnotation
[WARNING]   - org.jvnet.hk2.annotations.Optional
[WARNING]   - org.jvnet.hk2.annotations.Service
[WARNING] hk2-utils-2.6.1.jar, pulsar-presto-connector-original.jar define 4 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-utils/pom.properties
[WARNING]   - META-INF/maven/org.glassfish.hk2/hk2-utils/pom.xml
[WARNING]   - org.jvnet.hk2.component.MultiMap
[WARNING]   - org.jvnet.hk2.component.MultiMap$1
[WARNING] managed-ledger.jar, pulsar-presto-connector-original.jar define 299 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.apache.pulsar/managed-ledger/pom.properties
[WARNING]   - META-INF/maven/org.apache.pulsar/managed-ledger/pom.xml
[WARNING]   - MLDataFormats.proto
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$AddEntryCallback
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$ClearBacklogCallback
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$CloseCallback
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$DeleteCallback
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$DeleteCursorCallback
[WARNING]   - org.apache.bookkeeper.mledger.AsyncCallbacks$DeleteLedgerCallback
[WARNING]   - 289 more...
[WARNING] pulsar-metadata.jar, pulsar-presto-connector-original.jar define 54 overlapping classes and resources: 
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-metadata/pom.properties
[WARNING]   - META-INF/maven/org.apache.pulsar/pulsar-metadata/pom.xml
[WARNING]   - org.apache.pulsar.metadata.api.CacheGetResult
[WARNING]   - org.apache.pulsar.metadata.api.GetResult
[WARNING]   - org.apache.pulsar.metadata.api.MetadataCache
[WARNING]   - org.apache.pulsar.metadata.api.MetadataSerde
[WARNING]   - org.apache.pulsar.metadata.api.MetadataStore
[WARNING]   - org.apache.pulsar.metadata.api.MetadataStoreConfig
[WARNING]   - org.apache.pulsar.metadata.api.MetadataStoreConfig$MetadataStoreConfigBuilder
[WARNING]   - org.apache.pulsar.metadata.api.MetadataStoreException
[WARNING]   - 44 more...
[WARNING] javax.annotation-api-1.3.2.jar, pulsar-presto-connector-original.jar define 3 overlapping resources: 
[WARNING]   - META-INF/LICENSE.txt
[WARNING]   - META-INF/maven/javax.annotation/javax.annotation-api/pom.properties
[WARNING]   - META-INF/maven/javax.annotation/javax.annotation-api/pom.xml
[WARNING] jetty-io-9.4.48.v20220622.jar, jetty-security-9.4.27.v20200227.jar, jetty-server-9.4.48.v20220622.jar, jetty-servlet-9.4.48.v20220622.jar, jetty-util-9.4.48.v20220622.jar, jetty-util-ajax-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/NOTICE.txt
[WARNING]   - about.html
[WARNING] jetty-security-9.4.27.v20200227.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-security/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-security/pom.xml
[WARNING] javax.ws.rs-api-2.1.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/javax.ws.rs/javax.ws.rs-api/pom.properties
[WARNING]   - META-INF/maven/javax.ws.rs/javax.ws.rs-api/pom.xml
[WARNING] jetty-server-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 3 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-server/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-server/pom.xml
[WARNING]   - jetty-dir.css
[WARNING] jetty-servlet-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-servlet/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-servlet/pom.xml
[WARNING] jetty-util-ajax-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-util-ajax/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-util-ajax/pom.xml
[WARNING] jetty-util-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-util/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-util/pom.xml
[WARNING] jetty-io-9.4.48.v20220622.jar, pulsar-presto-connector-original.jar define 2 overlapping resources: 
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-io/pom.properties
[WARNING]   - META-INF/maven/org.eclipse.jetty/jetty-io/pom.xml
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://maven.apache.org/plugins/maven-shade-plugin/
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

I think I found the root cause. After the first run, we have both:

```
[DEBUG] Transforming META-INF/services/org.apache.pulsar.shade.org.glassfish.jersey.internal.inject.InjectionManagerFactory using org.apache.maven.plugins.shade.resource.ServicesResourceTransformer
[DEBUG] Transforming META-INF/services/org.glassfish.jersey.internal.inject.InjectionManagerFactory using org.apache.maven.plugins.shade.resource.ServicesResourceTransformer
```

... in the path. And perhaps due to `ServicesResourceTransformer` first treat them as different files, but after relocated, they are under the same name, so when shaded into the final jar file, it failed on:

```
Caused by: java.util.zip.ZipException: duplicate entry: META-INF/services/org.apache.pulsar.shade.org.glassfish.jersey.internal.inject.InjectionManagerFactory
    at java.util.zip.ZipOutputStream.putNextEntry (ZipOutputStream.java:241)
    at java.util.jar.JarOutputStream.putNextEntry (JarOutputStream.java:115)
    at org.apache.maven.plugins.shade.resource.ServicesResourceTransformer.modifyOutputStream (ServicesResourceTransformer.java:125)
    at org.apache.maven.plugins.shade.DefaultShader.shade (DefaultShader.java:155)
    at org.apache.maven.plugins.shade.mojo.ShadeMojo.execute (ShadeMojo.java:500)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

This should be an upstream issue to resolve. As long as we hold the dependency and relocation logic, it's always there. We may try to get rid of the dependency of `jersey-hk2-2.34.jar` and hope there're no more bad cases.

--- a/buildtools/pom.xml
+++ b/buildtools/pom.xml
@@ -43,7 +43,7 @@
     <slf4j.version>1.7.32</slf4j.version>
     <testng.version>7.3.0</testng.version>
     <commons-lang3.version>3.11</commons-lang3.version>
-    <maven-shade-plugin.version>3.2.4</maven-shade-plugin.version>
+    <maven-shade-plugin.version>3.4.0</maven-shade-plugin.version>
     <puppycrawl.checkstyle.version>8.37</puppycrawl.checkstyle.version>
     <maven-checkstyle-plugin.version>3.1.2</maven-checkstyle-plugin.version>
     <netty.version>4.1.77.Final</netty.version>

--- a/pom.xml
+++ b/pom.xml
@@ -260,7 +260,7 @@ flexible messaging model and an intuitive client API.</description>
     <maven-compiler-plugin.version>3.10.1</maven-compiler-plugin.version>
     <maven-dependency-plugin.version>3.3.0</maven-dependency-plugin.version>
     <maven-modernizer-plugin.version>2.3.0</maven-modernizer-plugin.version>
-    <maven-shade-plugin>3.3.0</maven-shade-plugin>
+    <maven-shade-plugin>3.4.0</maven-shade-plugin>
     <maven-antrun-plugin.version>3.0.0</maven-antrun-plugin.version>
     <properties-maven-plugin.version>1.0.0</properties-maven-plugin.version>
     <nifi-nar-maven-plugin.version>1.2.0</nifi-nar-maven-plugin.version>

